apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs

```diff
@@ -46,6 +116,159 @@ private SpannerConversionOptions(bool useDBNull)
         /// Determines the right conversion options to use based on the connection string of the given connection string builder.
         /// </summary>
         internal static SpannerConversionOptions ForConnectionStringBuilder(SpannerConnectionStringBuilder builder) =>
-            builder == null ? Default : builder.UseClrDefaultForNull ? s_useClrDefaultForNull : s_useDBNullForNull;
+            builder == null ? Default : builder.ConversionOptions;
+
+        internal SpannerConversionOptions Clone() => new SpannerConversionOptions(UseDBNull)
+        {
+            DateTimeToConfiguredSpannerType = DateTimeToConfiguredSpannerType,
+            DecimalToConfiguredSpannerType = DecimalToConfiguredSpannerType,
+            DateToConfiguredClrType = DateToConfiguredClrType,
+            Float64ToConfiguredClrType = Float64ToConfiguredClrType
+        };
+
+        internal SpannerConversionOptions WithClrDefaultForNullSetting(bool useClrDefaultToNull)
+        {
+            var clone = Clone();
+            clone.UseDBNull = !useClrDefaultToNull;
+            return clone;
+        }
+
+        internal SpannerConversionOptions WithClrToSpannerMappings(string clrToSpannerMappings)
+        {
+            var clone = Clone();
+            ValidateAndParseClrToSpannerTypeMappings(clrToSpannerMappings, clone);
+            return clone;
+        }
+
+        internal SpannerConversionOptions WithSpannerToClrMappings(string spannerToClrMappings)
+        {
+            var clone = Clone();
+            ValidateAndParseSpannerToClrTypeMappings(spannerToClrMappings, clone);
+            return clone;
+        }
+
+        private void ValidateAndParseSpannerToClrTypeMappings(string input, SpannerConversionOptions options)
+        {
+            // Empty string or null is valid to unset the values.
+            if (string.IsNullOrEmpty(input))
+            {
+                // Set defaults.
+                options.SetSpannerToClrTypeDefaults();
+                return;
+            }
+
+            // Get all type mappings.
+            var mappings = input.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I suspect we don't need `;` in the list here. (I don't *think* it should come through from the connection string builder.)
That would allow us to use `input.Split(',',StringSplitOptions.RemoveEmptyEntries)'`, avoiding allocating the extra array.